packages/agent-toolkit/src/utils/tools/tools-filtering.utils.ts

```diff
@@ -0,0 +1,82 @@
+import { ApiClient } from '@mondaydotcomorg/api';
+import { Tool, ToolSubType } from 'src/core';
+import { BaseMondayApiTool } from 'src/core/';
+import { ToolsConfiguration } from 'src/mcp';
+
+export function filterApiTools<T extends new (api: ApiClient) => BaseMondayApiTool<any>>(
+  tools: T[],
+  apiClient: ApiClient,
+  config?: ToolsConfiguration,
+): T[] {
+  if (!config) {
+    return tools;
+  }
+
+  // If dynamic API tools are enabled and read-only mode is not enabled, return all tools
+  let filteredTools = tools;
+  if (config.enableDynamicApiTools && !config.readOnlyMode) {
+    return filteredTools;
+  }
+
+  filteredTools = filteredTools.filter((tool) => {
+    const toolInstance = new tool(apiClient);
+    return toolInstance.subType !== ToolSubType.ALL_API;
+  });
+
+  if (config.include) {
+    filteredTools = tools.filter((tool) => {
+      const toolInstance = new tool(apiClient);
+      return config.include?.includes(toolInstance.name);
+    });
+  } else if (config.exclude) {
+    filteredTools = tools.filter((tool) => {
+      const toolInstance = new tool(apiClient);
+      return !config.exclude?.includes(toolInstance.name);
+    });
+  }
+
+  if (config.readOnlyMode) {
+    filteredTools = filteredTools.filter((tool) => {
+      const toolInstance = new tool(apiClient);
+      return toolInstance.subType === ToolSubType.READ;
+    });
+  }
+
+  return filteredTools;
+}
+
+export function filterMondayAppsTools<T extends new (...args: any[]) => Tool<any, any>>(
+  tools: T[],
+  mondayApiToken: string,
+  config?: ToolsConfiguration,
+): T[] {
+  if (!config) {
+    return tools;
+  }
+
+  if (!config.enableMondayAppsTools) {
+    return [];
+  }
+  let filteredTools = tools;
+
+  if (config.include) {
```

Comment:
I still don't understand why the filter logic is different, the "include", "exclude" and "readOnly" are exactly the same, why can't we have 1 filter method?